src/test/java/uk/gov/pay/directdebit/mandate/dao/SearchMandateDaoIT.java

```diff
@@ -0,0 +1,108 @@
+package uk.gov.pay.directdebit.mandate.dao;
+
+import junitparams.Parameters;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import uk.gov.pay.directdebit.DirectDebitConnectorApp;
+import uk.gov.pay.directdebit.junit.DropwizardConfig;
+import uk.gov.pay.directdebit.junit.DropwizardJUnitRunner;
+import uk.gov.pay.directdebit.junit.DropwizardTestContext;
+import uk.gov.pay.directdebit.junit.TestContext;
+import uk.gov.pay.directdebit.mandate.fixtures.MandateFixture;
+import uk.gov.pay.directdebit.mandate.model.Mandate;
+import uk.gov.pay.directdebit.payments.fixtures.GatewayAccountFixture;
+
+import java.util.List;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static uk.gov.pay.directdebit.mandate.fixtures.MandateFixture.aMandateFixture;
+import static uk.gov.pay.directdebit.mandate.params.MandateSearchParams.aMandateSearchParams;
+import static uk.gov.pay.directdebit.payments.fixtures.GatewayAccountFixture.aGatewayAccountFixture;
+
+@RunWith(DropwizardJUnitRunner.class)
+@DropwizardConfig(app = DirectDebitConnectorApp.class, config = "config/test-it-config.yaml")
+public class SearchMandateDaoIT {
+
+    @DropwizardTestContext
+    private TestContext testContext;
+
+    private MandateSearchDao mandateSearchDao;
+    private GatewayAccountFixture gatewayAccountFixture = aGatewayAccountFixture().withExternalId("gateway-account-id");
+    private MandateFixture mandate1 = aMandateFixture()
+            .withGatewayAccountFixture(gatewayAccountFixture)
+            .withServiceReference("REF1234");
+    private MandateFixture mandate2 = aMandateFixture()
+            .withGatewayAccountFixture(gatewayAccountFixture);
+    
+    @Before
+    public void setup() {
+        mandateSearchDao = new MandateSearchDao(testContext.getJdbi());
+
+        gatewayAccountFixture.insert(testContext.getJdbi());
+        mandate1.insert(testContext.getJdbi());
+        mandate2.insert(testContext.getJdbi());
+    }
+    
+    @Test
+    @Parameters({"REF1234", "ref1234", "f12"})
+    public void searchByReference(String searchString) {
+        var mandateSearchParams = aMandateSearchParams().withReference(searchString).withGatewayAccountId("gateway-account-id");
+        List<Mandate> results = mandateSearchDao.search(mandateSearchParams);
+        assertThat(results.size()).isEqualTo(1);
+        assertThat(results.get(0)).isEqualTo(mandate1.toEntity());
+    }
+    
+    @Test
+    public void searchByState() {
```

Comment:
What does this do?

Reply:
This would be quite useful in the event I don't finish the story before I go on holiday... someone can pick the story up and see that these tests needs to be implemented.